### PR TITLE
CLI custom bundle generation

### PR DESCRIPTION
**Pull request by mhipszki**
as described in #420 a CLI option has been introduced to allow users build their own bundles with a custom list of adapters

### CODE REVIEW

**Comment by mkendall07** [comment]
@protonate 
Interested to get your thoughts on this.

**Comment by mhipszki** [comment]
```
src/polyfill.js: line 6, col 18, Extending prototype of native object: 'Array'.

1 error
src/adapters/aardvark.js: line 27, col 11, 'ref' is already defined.
src/adapters/aardvark.js: line 48, col 121, 'ref' used out of scope.

2 errors
src/adapters/nginad.js: line 146, col 19, Don't make functions within a loop.

1 error
src/adapters/sonobi.js: line 42, col 52, Expected a 'break' statement before 'case'.

1 error
```

File: README.md [hunk 1]
```diff
@@ -114,15 +114,44 @@ If you experience errors, after a version update, try a fresh install:
...
+- `./build/dev/prebid.js` Full source code for dev and debug
+- `./build/dev/prebid.js.map` Source map for dev and debug
+- `./build/dist/prebid.js` Minified production code
+- `./prebid.js_<version>.zip` Distributable
 
 Code quality is defined by `./.jscs` and `./.jshint` files and errors are reported in the
 terminal. The build will continue with quality errors, however. If you are contributing code
 you can configure your editor with the provided .jscs and .jshint settings.
 
+### Optimisation ###
```
Comment (mkendall07) [inline README.md]: Looks good. Could you break these into 2 sections maybe? I'm thinking something like 1) Build standalone prebid.js 2) Build using NPM for bundling

File: loaders/adapterLoader.js [hunk 2]
```diff
@@ -5,9 +5,11 @@
 
 'use strict';
 
+
 const fs = require('fs');
 const blockLoader = require('block-loader');
-const adapters = require('../package.json').adapters;
+
```
Comment (mkendall07) [inline loaders/adapterLoader.js]: please remove spaces.

File: loaders/getAdapters.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const argv = require('yargs').argv;
+
+const jsonPath = argv['adapters'] || '';
+
+module.exports = function getAdapters(all) {
+    const json = path.resolve(process.cwd(), jsonPath);
+
+    let adapters;
+    try {
+        const stat = fs.statSync(json);
```
Comment (mkendall07) [inline loaders/getAdapters.js]: `stat` unused.

File: loaders/getAdapters.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+'use strict';
```
Comment (mkendall07) [inline loaders/getAdapters.js]: please `jshcs` fix this file.

**Comment by mkendall07** [comment]
@mhipszki 
This looks great. Just a few minor things I commented on. Once they are fixed I can merge. Thanks

File: README.md (hunk 1 above)
Comment (mhipszki) [inline README.md]: done

File: loaders/adapterLoader.js (hunk 2 above)
Comment (mhipszki) [inline loaders/adapterLoader.js]: done

also added `.editorconfig` rules to the project (to help contributors submitting changes in consistent style)

File: loaders/getAdapters.js (hunk 3 above)
Comment (mhipszki) [inline loaders/getAdapters.js]: fixed

File: loaders/getAdapters.js (hunk 4 above)
Comment (mhipszki) [inline loaders/getAdapters.js]: not sure about this?

**Comment by matthewlane** [comment]
@mhipszki Thanks for this. Would you mind dropping the .editorconfig from this PR? We'd like to learn more about .editorconfig and consider it in a separate pull

**Comment by mhipszki** [comment]
@matthewlane sure, you're right actually `.editorconfig` should not be part of this PR, removed

**Comment by mhipszki** [comment]
@mkendall07 @matthewlane I've merged in the latest master, this one is ready to go now ;)

**Comment by mkendall07** [comment]
LGTM. Thanks for contributing

**Comment by mhipszki** [comment]
@mkendall07 awesome, thanks for merging! :)

**Pull request merged**